### PR TITLE
Fix concurrency issue in RaptorSplitSource

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by dain** [comment]
needs a better commit message

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java [hunk 1]
```diff
@@ -171,6 +179,23 @@ public boolean isFinished()
...
 
+        private Supplier<List<ConnectorSplit>> batchSupplier(int maxSize)
+        {
+            return () -> {
+                ImmutableList.Builder<ConnectorSplit> list = ImmutableList.builder();
+                for (int i = 0; i < maxSize; i++) {
+                    if (Thread.currentThread().isInterrupted()) {
+                        throw new PrestoException(INTERNAL_ERROR, "Split batch fetch was interrupted");
+                    }
+                    if (!iterator.hasNext()) {
```
Comment (dain) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java]: not sure if you care, but both hasNext and next may throw an illegal state now

**Comment by dain** [comment]
looks good

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java [hunk 2]
```diff
@@ -140,11 +140,13 @@ public ConnectorSplitSource getPartitionSplits(ConnectorSession session, Connect
         private final TupleDomain<RaptorColumnHandle> effectivePredicate;
         private final CloseableIterator<ShardNodes> iterator;
 
+        private CompletableFuture<List<ConnectorSplit>> future;
```
Comment (martint) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java]: Is this class meant to be thread-safe? This non-final field is initialized and accessed without any protection

Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java]: `ConnectorSplitSource` should not need to be thread safe (from talking to @dain and my brief review of `SourcePartitionedScheduler` and `SqlQueryScheduler`).

Comment (martint) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java]: SourcePartitionedScheduler seems to try to be thread-safe (`schedule` is synchronized), but `close` (which delegates to `splitSource` is not). So, to me, that would seem to require that splitSource be thread-safe (not necessarily *concurrent-safe*)

**Pull request merged**